### PR TITLE
Milestone 2: App Structure

### PR DESCRIPTION
**Pull request by mohammadumar28**
* Create the basic components
* Create the reducer and actions functions
* Initialize state in the index.js
* Connect all the components to the state

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const CREATE_BOOK = 'CREATE_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
+
+function createBook(book) {
```
Comment (christianotieno) [inline src/actions/index.js:4]: ## Implement ES6 syntax
- [ ] Implement this `createBook` action with ES6 arrow function.
```suggestion
const createBook = (book) => {
```

File: src/actions/index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  return {
+    type: CREATE_BOOK,
+    book,
+  };
+}
+
+function removeBook(book) {
```
Comment (christianotieno) [inline src/actions/index.js:11]: ## Implement ES6 syntax
- [ ] Implement this `removeBook` action with ES6 arrow function.
```suggestion
const removeBook = (book) => {
```

File: src/components/App.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import BooksList from './BooksList';
+import BooksForm from './BooksForm';
+
+function App() {
```
Comment (christianotieno) [inline src/components/App.js:4]: ## Implement ES6 syntax
- [ ] Implement this `App` component with ES6 arrow function.
```suggestion
const App = () => {
```

File: src/components/Book.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import PropTypes from 'prop-types';
+
+function Book({ book }) {
```
Comment (christianotieno) [inline src/components/Book.js:3]: ## Implement ES6 syntax
- [ ] Implement this `Book` component with ES6 arrow function.
```suggestion
const Book = ({ book }) => {
```

File: src/components/BooksForm.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+function BooksForm() {
```
Comment (christianotieno) [inline src/components/BooksForm.js:1]: ## Implement ES6 syntax
- [ ] Implement this `BooksForm` component with ES6 arrow function.
```suggestion
const BooksForm = () => {
```

File: src/components/BooksList.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Book from './Book';
+
+function mapStateToProps(state) {
```
Comment (christianotieno) [inline src/components/BooksList.js:6]: ## Implement ES6 syntax
- [ ] Implement this `mapStateToProps` function's component with ES6 arrow function.
```suggestion
const mapStateToProps = (state) => {
```

File: src/components/BooksList.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  return {
+    books: state.books,
+  };
+}
+
+function BooksList({ books }) {
```
Comment (christianotieno) [inline src/components/BooksList.js:12]: ## Implement ES6 syntax
- [ ] Implement this `BooksList` component with ES6 arrow function.
```suggestion
const BooksList = ({ books }) => {
```

File: src/reducers/books.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+function reducer(state = {}, action) {
```
Comment (christianotieno) [inline src/reducers/books.js:1]: ## Implement ES6 syntax
- [ ] Implement this `reducer` reducer with ES6 arrow function.
```suggestion
const reducer = (state = {}, action) => {
```
## Implement the `index.js` reducer
- [ ] While at it please ensure you have implemented the `index.js` combine reducer that required to combine your app's reducers. As of now, it is currently empty.

**Review by christianotieno — CHANGES_REQUESTED** [review]
## Changes Required ♻️ 
Hello @mohammadumar28 👋 
I trust you are doing well. 
Good job with this milestone, we have some few changes to work on before we proceed.
I have left some comments down below 👇 

### Happy coding 💻 
#### Christian Otieno (CR)

**Review by whiz25 — APPROVED** [review]
## Approved :clap:
Changes made as requested. Syntax follow ES6 syntax. Nice job.
Merge and proceed to the next milestone.


### Martin Afani

**Pull request merged**